tests/test_mlkit.py

```diff
@@ -0,0 +1,132 @@
+
+ERROR_CODE = 404
+ERROR_MSG = 'The resource was not found'
+ERROR_STATUS = 'NOT_FOUND'
+ERROR_JSON = {
+    'error': {
+        'code': ERROR_CODE,
+        'message': ERROR_MSG,
+        'status': ERROR_STATUS
+    }
+}
+_ERROR_RESPONSE = json.dumps(ERROR_JSON)
+
+
+class TestGetModel(object):
+    """Tests mlkit.get_model."""
+    @classmethod
+    def setup_class(cls):
+        cred = testutils.MockCredential()
+        firebase_admin.initialize_app(cred, {'projectId': PROJECT_ID})
+
+    @classmethod
+    def teardown_class(cls):
+        testutils.cleanup_apps()
+
+    @staticmethod
+    def check_error(err, errType, msg):
+        assert isinstance(err, errType)
+        assert str(err) == msg
+
+    @staticmethod
+    def check_firebase_error(err, code, status):
+        assert isinstance(err, exceptions.FirebaseError)
+        assert err.code == code
+        assert err.http_response is not None
+        assert err.http_response.status_code == status
+
+    def _get_url(self, project_id, model_id):
+        return mlkit._MLKitService.BASE_URL + 'projects/{0}/models/{1}'.format(project_id, model_id)
+
+    def _instrument_mlkit_service(self, app=None, status=200, payload=_DEFAULT_RESPONSE):
+        if not app:
+            app = firebase_admin.get_app()
+        mlkit_service = mlkit._get_mlkit_service(app)
+        recorder = []
+        mlkit_service._client.session.mount(
+            'https://mlkit.googleapis.com',
+            testutils.MockAdapter(payload, status, recorder)
+        )
+        return mlkit_service, recorder
+
+    def test_get_model(self):
+        _, recorder = self._instrument_mlkit_service()
+        model = mlkit.get_model(MODEL_ID_1)
+        assert len(recorder) == 1
+        assert recorder[0].method == 'GET'
+        assert recorder[0].url == self._get_url(PROJECT_ID, MODEL_ID_1)
+        assert model == MODEL_1
+        assert model._data['name'] == MODEL_NAME_1
+        assert model._data['displayName'] == DISPLAY_NAME_1
+
+    def test_get_model_validation_errors(self):
+        _, recorder = self._instrument_mlkit_service()
+        #Empty model-id
+        with pytest.raises(ValueError) as err:
+            mlkit.get_model('')
+        self.check_error(err.value, ValueError, 'Model Id is required for GetModel.')
+
+        #Wrong type
+        with pytest.raises(TypeError) as err:
+            mlkit.get_model(12345)
+        self.check_error(err.value, TypeError, 'Model Id must be a string.')
+
+        #Invalid characters
+        with pytest.raises(ValueError) as err:
+            mlkit.get_model('&_*#@:/?')
+        self.check_error(err.value, ValueError, 'Model Id format is invalid.')
+
+    def test_get_model_error(self):
+        _, recorder = self._instrument_mlkit_service(status=404, payload=_ERROR_RESPONSE)
+        with pytest.raises(exceptions.NotFoundError) as err:
+            mlkit.get_model(MODEL_ID_1)
+        self.check_firebase_error(err.value, ERROR_STATUS, ERROR_CODE)
```

Comment:
Also assert the error message. If the OP error got parsed correctly, error message gets set to the `error.message` field of the response.

Reply:
done